extensions/amp-base-carousel/1.0/base-carousel.js

```diff
@@ -49,21 +47,32 @@ export function BaseCarousel({
     carouselContext.setCurrentSlide ?? setCurrentSlideState;
   const {setSlideCount} = carouselContext;
   const scrollRef = useRef(null);
-  const advance = (by) => scrollRef.current.advance(by);
-  useMountEffect(() => {
-    if (setAdvance) {
-      setAdvance(advance);
-    }
-  });
+
+  const advance = useCallback((by) => scrollRef.current.advance(by), []);
+  const setRestingIndex = useCallback(
```

Comment:
why is it called `setRestingIndex` vs. `setSlideIndex`?

Reply:
nit: if the only side effect is to call `onSlideChange`, it might be cleaner to inline it in the exposed `goToSlide` fn definition instead of creating a new callback.

Reply:
Might be. Not sure. I was more targeted in this pull request to only change imperative API. Maybe @caroqliu can comment on this, but I think there are other needs for `setRestingIndex` in this shape.

Reply:
[This is where the name comes from](https://github.com/ampproject/amphtml/blob/a6e1e1c53bec81287aacb937fc9dcf319cca79af/extensions/amp-base-carousel/1.0/scroller.js#L171-L173) as it is both naming the method by what it's doing and the "concept" of a `restingIndex` in the carousel semantics. Happy to rename the former as long as it's in sync with the latter.

Reply:
As an additional point of clarification, the order we want to display the slides is not always the same as the slide order in the DOM, as a result of the `loop` feature.

Reply:
IMHO the naming is ok if it makes sense to the component. I think this is just a point of clarification.